test/SwapRouter.fork.8453.t.sol.sol

```diff
@@ -0,0 +1,289 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {Test, console2} from "forge-std/Test.sol";
+import {SwapRouter} from "src/SwapRouter.sol";
+import {IERC20, ERC20} from "openzeppelin/token/ERC20/ERC20.sol";
+import {CreatorToken} from "src/CreatorToken.sol";
+import {MockIncrementingBondingCurve} from "test/mocks/MockIncrementingBondingCurve.sol";
+import {IQuoterV2} from "test/interfaces/IQuoterV2.sol";
+
+contract SwapRouterTest is Test {
+  uint256 baseFork;
+
+  SwapRouter router;
+  address constant WETH_ADDRESS = 0x4200000000000000000000000000000000000006;
+  address constant USDC_ADDRESS = 0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913;
+  uint256 constant BASE_PAY_AMOUNT = 1e6; // Because USDC has 6 decimals
+  uint24 constant LOW_FEE_TIER = 500;
+  string CREATOR_TOKEN_NAME = "Test Token";
+  string CREATOR_TOKEN_SYMBOL = "TEST";
+  string CREATOR_TOKEN_URI = "URI";
+
+  address quoterV2Address = 0x3d4e44Eb1374240CE5F1B871ab261CD16335B76a; // On base
+  address creator = vm.addr(1);
+  address admin = vm.addr(2);
+  address referrer;
+  uint256 creatorFee;
+  uint96 creatorRoyalty;
+  uint256 adminFee;
+
+  ERC20 payToken;
+  MockIncrementingBondingCurve bondingCurve;
+  CreatorToken creatorToken;
+
+  function setUp() public {
+    string memory BASE_RPC_URL = vm.envString("BASE_RPC_URL");
+    baseFork = vm.createSelectFork(BASE_RPC_URL, 5_129_051);
+    // vm.rollFork(5_129_051);
+
+    payToken = ERC20(USDC_ADDRESS); // USDC on base
+    bondingCurve = new MockIncrementingBondingCurve(BASE_PAY_AMOUNT);
+    creatorToken =
+    new CreatorToken(CREATOR_TOKEN_NAME, CREATOR_TOKEN_SYMBOL, CREATOR_TOKEN_URI, creator, creatorFee, creatorRoyalty, admin, adminFee, referrer, payToken, bondingCurve);
+    router = new SwapRouter();
+
+    vm.label(address(payToken), "payToken contract");
+    vm.label(address(bondingCurve), "bondingCurve contract");
+    vm.label(address(creatorToken), "creatorToken contract");
+    vm.label(address(router), "router contract");
+  }
+
+  // Check Uniswap Docs: https://docs.uniswap.org/contracts/v3/reference/periphery/lens/QuoterV2
+  function quote(uint256 _amountOut) public returns (uint256 _amountIn) {
+    // struct QuoteExactOutputSingleParams {
+    //     address tokenIn;
+    //     address tokenOut;
+    //     uint256 amount;
+    //     uint24 fee;
+    //     uint160 sqrtPriceLimitX96;
+    // }
+    IQuoterV2.QuoteExactOutputSingleParams memory params = IQuoterV2.QuoteExactOutputSingleParams(
+      WETH_ADDRESS, USDC_ADDRESS, _amountOut, LOW_FEE_TIER, 0
+    );
+    (_amountIn,,,) = IQuoterV2(quoterV2Address).quoteExactOutputSingle(params);
+  }
+
+  function _assumeSafeBuyer(address _buyer) public view {
+    vm.assume(
+      _buyer != address(0) && _buyer != address(creatorToken) && _buyer != creator
+        && _buyer != admin && _buyer != referrer
+    );
+  }
+
+  function test_BuyWithEth(address _buyer) public {
```

Comment:
Some of our tests should intentionally send some extra ETH, then assert that the ETH has been returned to the buyer at the end of the transaction.

Reply:
Updated some tests here: [ced1396](https://github.com/showtime-xyz/creator-tokens/pull/44/commits/ced1396d07f469ee1a09006a368a8c1ebedb91ac)